README.md

```diff
@@ -28,7 +28,7 @@ if you want to write your own extensions, whether for new Node implementations
 or new codecs, or new higher-order order functions!)
 
 - `github.com/ipld/go-ipld-prime` -- imported as just `ipld` -- contains the core interfaces for IPLD.  The most important interfaces are `Node`, `NodeBuilder`, `Path`, and `Link`.
-- `github.com/ipld/go-ipld-prime/node/basic` -- imported as `basicnode` -- provides concrete implementations of `Node` and `NodeBuilder` which work for any kind of data.
+- `github.com/ipld/go-ipld-prime/node/basicnode` -- imported as `basicnode` -- provides concrete implementations of `Node` and `NodeBuilder` which work for any kind of data.
```

Comment:
This package name change will mean that this is an update that cannot be upgraded over transparently in downstream libraries.

Reply:
Yeah, that's the one thing in here that's unequivocally awful.

I think... I might actually propose we add back in a transition package, while leaving this move, and all the docs changes, in.

There are 17 symbols to wrap:

```
var Prototype prototype
func Chooser(_ datamodel.Link, _ linking.LinkContext) (datamodel.NodePrototype, error)
func NewBool(value bool) datamodel.Node
func NewBytes(value []byte) datamodel.Node
func NewFloat(value float64) datamodel.Node
func NewInt(value int64) datamodel.Node
func NewLink(value datamodel.Link) datamodel.Node
func NewString(value string) datamodel.Node
type Prototype__Any struct{}
type Prototype__Bool struct{}
type Prototype__Bytes struct{}
type Prototype__Float struct{}
type Prototype__Int struct{}
type Prototype__Link struct{}
type Prototype__List struct{}
type Prototype__Map struct{}
type Prototype__String struct{}
```

7 functions, 1 variable, 9 aliases.  That's not bad.  We can do that as a transitional measure, buying us a transitional time to remove them from downstreams lazily (which is also "sed refactor" complexity territory, i.e. near zero).

@mvdan thoughts on this as migration strategy?  (It was your naming review and preference in `bindnode` that made me want to do this, also.)

Reply:
This turned out to be extremely easy to do and will be included in another commit here soon.